tests/core/test_tracker_stores.py

```diff
@@ -976,3 +982,328 @@ def test_create_awaitable_tracker_store_with_endpoint_config():
+    assert tracker.events == deque()
+
+
+async def test_sql_get_or_create_full_tracker_with_action_listen() -> None:
+    tracker_store = SQLTrackerStore(Domain.empty())
+    sender_id = uuid.uuid4().hex
+    tracker = await tracker_store.get_or_create_full_tracker(
+        sender_id=sender_id, append_action_listen=True
+    )
+    assert tracker.sender_id == sender_id
+    assert tracker.events == deque([ActionExecuted(ACTION_LISTEN_NAME)])
+
+
+async def test_sql_get_or_create_full_tracker_with_existing_tracker(
+    tracker_with_restarted_event: DialogueStateTracker,
+) -> None:
+    sender_id = tracker_with_restarted_event.sender_id
+
+    tracker_store = SQLTrackerStore(Domain.empty())
+    await tracker_store.save(tracker_with_restarted_event)
+
+    tracker = await tracker_store.get_or_create_full_tracker(
+        sender_id=sender_id, append_action_listen=False
+    )
+    assert tracker.sender_id == sender_id
+    assert tracker.events == deque(tracker_with_restarted_event.events)
+
+
+async def test_sql_tracker_store_retrieve_full_tracker(
+    domain: Domain, tracker_with_restarted_event: DialogueStateTracker
+) -> None:
+    tracker_store = SQLTrackerStore(domain)
+    sender_id = tracker_with_restarted_event.sender_id
+    await tracker_store.save(tracker_with_restarted_event)
+
+    tracker = await tracker_store.retrieve_full_tracker(sender_id)
+    assert tracker == tracker_with_restarted_event
+
+
+async def test_sql_tracker_store_retrieve(
+    domain: Domain,
+    tracker_with_restarted_event: DialogueStateTracker,
+) -> None:
+    tracker_store = SQLTrackerStore(domain)
+    sender_id = tracker_with_restarted_event.sender_id
+    await tracker_store.save(tracker_with_restarted_event)
+
+    split_conversations = rasa.shared.core.events.split_events(
+        tracker_with_restarted_event.events,
+        ActionExecuted,
+        {"action_name": ACTION_SESSION_START_NAME},
+        include_splitting_event=False,
+    )
+    events_after_restart = split_conversations[-1]
+
+    tracker = await tracker_store.retrieve(sender_id)
+    assert list(tracker.events) == events_after_restart
+
+
+async def test_in_memory_tracker_store_retrieve_full_tracker(
+    domain: Domain,
+    tracker_with_restarted_event: DialogueStateTracker,
+) -> None:
+    tracker_store = InMemoryTrackerStore(domain)
+    sender_id = tracker_with_restarted_event.sender_id
+
+    await tracker_store.save(tracker_with_restarted_event)
+
+    tracker = await tracker_store.retrieve_full_tracker(sender_id)
+    assert tracker == tracker_with_restarted_event
+
+
+async def test_in_memory_tracker_store_retrieve(
+    domain: Domain,
+    tracker_with_restarted_event: DialogueStateTracker,
+) -> None:
+    tracker_store = InMemoryTrackerStore(domain)
+    sender_id = tracker_with_restarted_event.sender_id
+    await tracker_store.save(tracker_with_restarted_event)
+
+    split_conversations = rasa.shared.core.events.split_events(
```

Comment:
Can we replace this with the raw list of items?
It would be easier to read what the expected result should be.